rfcs/text/0352-graph-type-system.md

```diff
@@ -0,0 +1,2586 @@
+
+The `links` keyword specifically allows constraints in the case of having a set of links. Here the link set can be ordered or unordered (default).
+
+As mentioned in the Guide-Level explanation, `description` for Link Types add semantic meaning to a link, which could be encapsulated in a vocabulary specification for the meta schemas.
+
+For the most part, we're using existing JSON Schema keywords, but it would be preferable to look into defining these additions as Vocabularies or otherwise integrate them into a JSON Schema validator used for the proposed system.
+
+## Using the Types in the Block Protocol
+
+Using the proposed type system for Block Protocol imposes changes on the Graph Service and how Block Schemas are defined.
+
+The Graph Service is currently specified as [this schema](https://github.com/blockprotocol/blockprotocol/blob/main/packages/%40blockprotocol/graph/src/graph-service.json).
+
+### Interfacing with properties on Entities
+
+A key change for allowing the proposed type system to work is moving away from arbitrary property keys and making use of canonical property URIs. As seen in the examples in earlier sections, properties that use simple keys will now have to point at _Property Type URIs_.
+
+**An example of an entity instance in the current system:**
+
+```json
+{
+  "entityId": 111,
+  "name": "Arthur Philip Dent",
+  "age": 30,
+  "planetOfOrigin": "Earth",
+  "occupation": "Intergalactic Traveler"
+}
+```
+
+**An example of an entity instance in the proposed system:**
+
+```json
+{
+  "https://blockprotocol.org/types/@blockprotocol/property-type/entity-id": 111,
+  "https://blockprotocol.org/types/@alice/property-type/name": "Arthur Philip Dent",
+  "https://blockprotocol.org/types/@alice/property-type/age": 30,
+  "https://blockprotocol.org/types/@alice/property-type/planet-of-origin": "Earth",
+  "https://blockprotocol.org/types/@alice/property-type/occupation": "Intergalactic Traveler"
+}
+```
+
+This change canonicalizes property keys, such that they uniquely identify a Property Type.
+
+**Within blocks** this impacts:
+
+- how entity fields are accessed
+- how Block Protocol messages are constructed
+- payload size of Block Protocol messages
+
+**For block authors** this impacts:
+
+- how Block Schemas are defined
+- reusability of properties (the need for some kind of type discovery)
+
+> 💡 Notice that `entityId`s are replaced by a Property Type URI that resides in the `/@blockprotocol` namespace. This is an implementation detail, and not something that is strictly dictated by the proposal.
+
+### Interfacing with Linked Entities
+
+#### Receiving Links and Linked Entities
+
+Link groups and Linked entities in the Graph Service are currently supplied outside the entity as separate objects. This behavior will stay the same, but the objects received will be of a different shape.
+
+**An example of a `linkGroup` instance in the current system:**
```

Comment:
It would be the item in the `linkGroups` array, yes, but the type there is `$ref`d to be [/graph/link-group](https://blockprotocol.org/types/services/graph/link-group) in [the spec](https://github.com/blockprotocol/blockprotocol/blob/main/packages/%40blockprotocol/graph/src/graph-service.json#L773) which is what's being referred to here. 

I think it would be really nice to be able to link directly to these directly, I was just unaware that the types were live yet. I might go through the examples in this section and add direct links to them

Reply:
Unrelated, i think the `current` example is wrong here, whoopsie!